### PR TITLE
Enable uploading cse logs to storage account

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Allows specifying parameters in the deployment to upload cse logs to a storage account at the end of cse script. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: parts/k8s/kubernetescustomscript.sh [hunk 1]
```diff
@@ -496,6 +496,25 @@ configAddons() {
...
+installAzCli(){
+    AZ_REPO=$(lsb_release -cs)
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
+    sudo tee /etc/apt/sources.list.d/azure-cli.list
+    retrycmd_if_failure 3 1 30 curl -L https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+    apt_get_update && apt_get_install 2 20 120 azure-cli
+}
+
+exitScript() {
+    if [[ "$SAS_TOKEN" != "" ]]; then
```
Comment (weinong) [inline parts/k8s/kubernetescustomscript.sh]: can we only upload when all the required env vars are set?

Comment (CecileRobertMichon) [reply parts/k8s/kubernetescustomscript.sh]: fixed

File: parts/k8s/kubernetescustomscript.sh [hunk 2]
```diff
@@ -496,6 +496,25 @@ configAddons() {
     fi
 }
 
+installAzCli(){
+    AZ_REPO=$(lsb_release -cs)
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
+    sudo tee /etc/apt/sources.list.d/azure-cli.list
+    retrycmd_if_failure 3 1 30 curl -L https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: The two-step approach elsewhere (get artifact / add to key) is more resilient to network flakiness

File: parts/k8s/kubernetescustomscript.sh [hunk 3]
```diff
@@ -496,6 +496,25 @@ configAddons() {
...
+    apt_get_update && apt_get_install 2 20 120 azure-cli
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: Any reason why only 2 retries?

File: parts/k8s/kubernetescustomscript.sh [hunk 4]
```diff
@@ -496,6 +496,25 @@ configAddons() {
...
+}
+
+exitScript() {
+    if [[ "$SAS_TOKEN" != "" ]] && [[ "$CONTAINER_NAME" != "" ]] && [[ "$STORAGE_ACCOUNT_NAME" != "" ]] && [[ "$BLOB_PREFIX" != "" ]]; then
+        installAzCli || echo "Failed to install az cli"
+        set +x
+        az storage blob upload -f /var/log/azure/cluster-provision.log -c $CONTAINER_NAME -n $BLOB_PREFIX-`hostname` --account-name $STORAGE_ACCOUNT_NAME --sas-token $SAS_TOKEN || echo "Failed to upload to Storage Account"
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: I guess if the installAzCli fails above we'll just hit the or condition here on file not found?

File: parts/k8s/kubernetescustomscript.sh (hunk 3 above)
Comment (CecileRobertMichon) [reply parts/k8s/kubernetescustomscript.sh]: The thought here was: we don't want to delay/block deployment for logs so being resilient in this step doesn't matter a lot

File: parts/k8s/kubernetescustomscript.sh (hunk 2 above)
Comment (jackfrancis) [reply parts/k8s/kubernetescustomscript.sh]: Let's reduce the timeout to 5 seconds and retry, say, 10 times

File: parts/k8s/kubernetescustomscript.sh (hunk 3 above)
Comment (jackfrancis) [reply parts/k8s/kubernetescustomscript.sh]: Let's reduce timeout to 60 and retry 3 times

**Pull request closed without merging**